wsgi.py

```diff
@@ -95,6 +105,15 @@ def post_provenance_result():  # Ignore PyDocStyleBear
     adapter = ProvenanceResultsStore()
     adapter.connect()
     document_id = adapter.store_document(request.json)
+    if(request.form.get("origin")):
+        url = request.form.get("origin")
+        service = url.split("/")[2].split(".")[0]
```

Comment:
Might be nitpicking, but worth to name this as a separate method and reuse this in both scenarios:

```python
def _get_service_from_origin(origin: str) -> str:
    return url.split("/")[2].split(".")[0]
```